trestle/transforms/implementations/osco.py

```diff
@@ -577,66 +577,107 @@ def __init__(
 
     def transform(self, profile: Profile) -> str:
         """Transform the Profile into a OSCO yaml."""
+        self._profile = profile
+        self._check_version = self._get_normalized_version('profile_check_version', '0.1.59')
+        self._osco_version = self._get_normalized_version('profile_osco_version', '0.1.46')
         # set values
-        set_values = self._get_set_values(profile)
+        set_values = self._get_set_values()
         # spec
-        spec = {
-            'description': self._get_metadata_prop_value(profile, 'profile_mnemonic', self._name),
-            'extends': self._get_metadata_prop_value(profile, 'base_profile_mnemonic', self._extends),
-            'title': profile.metadata.title,
-            'setValues': set_values,
-        }
-        disable_rules = self._get_disable_rules(profile)
+        if self._osco_version < 1046:
+            # for versions prior to 0.1.46, exclude 'description'
+            spec = {
+                'extends': self._get_metadata_prop_value('base_profile_mnemonic', self._extends),
+                'title': self._profile.metadata.title,
+                'setValues': set_values,
+            }
+        else:
+            # for versions 0.1.46 and beyond, include 'description'
+            spec = {
+                'description': self._get_metadata_prop_value('profile_mnemonic', self._name),
+                'extends': self._get_metadata_prop_value('base_profile_mnemonic', self._extends),
+                'title': self._profile.metadata.title,
+                'setValues': set_values,
+            }
+        disable_rules = self._get_disable_rules()
         if len(disable_rules) > 0:
             spec['disableRules'] = disable_rules
         # yaml data
         ydata = {
             'apiVersion': self._api_version,
             'kind': self._kind,
             'metadata': {
-                'name': self._get_metadata_prop_value(profile, 'profile_mnemonic', self._name),
+                'name': self._get_metadata_prop_value('profile_mnemonic', self._name),
                 'namespace': self._namespace,
             },
             'spec': spec,
         }
         return json.dumps(ydata)
 
-    def _get_set_values(self, profile) -> List[Dict]:
+    def _get_normalized_version(self, prop_name, prop_default) -> None:
+        """Get normalized version.
+
+        Normalize the "x.y.z" string value to an integer: 1,000,000*x + 1,000*y + z.
+        """
+        try:
+            vparts = self._get_metadata_prop_value(prop_name, prop_default).split('.')
+            normalized_version = int(vparts[0]) * 1000000 + int(vparts[1]) * 1000 + int(vparts[2])
+        except Exception:
+            logger.warning(f'metadata prop name={prop_name} value error')
```

Comment:
If you know it is of the form x.y.z you can just sort and compare based on the tuple (x, y, z), where each is the corresponding integer.  Not a big deal but lets you treat everything as strings except in the compare.

Reply:
Great suggestion! Fixed.